# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/CanvasTranslator.cpp
# Commit: 12aa9be1daa2
# Full Hash: 12aa9be1daa25a0f455c799f62cb5391971792d3
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2023-12-10 09:36:11
# Description:
#   Bug 1869085 - Make CanvasManagerParent use content parent ID for recording textures. r=bobowen
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195957
# ==============================================================================

diff -r 19967878c4b0 -r 12aa9be1daa2 gfx/layers/ipc/CanvasTranslator.cpp
--- a/gfx/layers/ipc/CanvasTranslator.cpp	Sat Dec 09 19:49:22 2023 +0000
+++ b/gfx/layers/ipc/CanvasTranslator.cpp	Sat Dec 09 20:16:02 2023 +0000
@@ -50,8 +50,11 @@
   return textureData;
 }
 
-CanvasTranslator::CanvasTranslator() {
-  mMaxSpinCount = StaticPrefs::gfx_canvas_remote_max_spin_count();
+CanvasTranslator::CanvasTranslator(const dom::ContentParentId& aContentId,
+                                   uint32_t aManagerId)
+    : mMaxSpinCount(StaticPrefs::gfx_canvas_remote_max_spin_count()),
+      mContentId(aContentId),
+      mManagerId(aManagerId) {
   mNextEventTimeout = TimeDuration::FromMilliseconds(
       StaticPrefs::gfx_canvas_remote_event_timeout_ms());
 
@@ -190,6 +193,7 @@
                                  size_t aBufferSize) {
   MOZ_ASSERT(IsInTaskQueue());
   MOZ_RELEASE_ASSERT(mHeader->readerState == State::Paused);
+  MOZ_ASSERT(mDefaultBufferSize != 0);
 
   // Default sized buffers will have been queued for recycling.
   if (mCurrentShmem.Size() == mDefaultBufferSize) {
@@ -302,6 +306,7 @@
 
 void CanvasTranslator::FinishShutdown() {
   MOZ_ASSERT(gfx::CanvasRenderThread::IsInCanvasRenderThread());
+  gfx::CanvasManagerParent::RemoveReplayTextures(this);
 }
 
 bool CanvasTranslator::CheckDeactivated() {